File: implementations/java-springboot/examples/src/main/java/com/example/accountservice/account/application/service/NlTransactionQueryTranslator.java
package com.example.accountservice.account.application.service;

/**
 * A Technical Service (see root docs/architecture/domain-service.md) translating a free-text
 * question about an account's transaction history into a structured filter. This is the
 * "Retrieve"-preparation step of a structured-data RAG pipeline: {@link
 * NlTransactionAnswerComposer} (the "Generate" step) is the other half, and {@code
 * AccountQuery.findTransactions} itself is the "Retrieve" step in between.
 */
public interface NlTransactionQueryTranslator {

    TransactionFilter translate(String question);
}

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/accountservice/account/application/service/NlTransactionQueryTranslator.kt
package com.example.accountservice.account.application.service

/**
 * A Technical Service (see root docs/architecture/domain-service.md) translating a free-text
 * question about an account's transaction history into a structured filter. This is the
 * "Retrieve"-preparation step of a structured-data RAG pipeline: [NlTransactionAnswerComposer] (the
 * "Generate" step) is the other half, and `AccountQuery.findTransactions` itself is the "Retrieve"
 * step in between.
 */
interface NlTransactionQueryTranslator {
    fun translate(question: String): TransactionFilter
}
